neuralmonkey/vocabulary.py

```diff
@@ -352,6 +357,24 @@ def trunkate(self, size: int) -> None:
         for index, word in enumerate(self.index_to_word):
             self.word_to_index[word] = index
 
+    def truncate_by_min_freq(self, min_freq: int) -> None:
+        """Truncate the vocabulary by only keeping tokens with given minimum
+        frequency.
+
+        Arguments:
+            min_freq: The minimum frequency of included words.
+        """
+        if min_freq == 1:
```

Comment:
This will fail if `min_freq` is `None`. Either remove the `Optional` type hint or add `is not None` to the condition.